src/vs/workbench/contrib/terminalContrib/links/browser/terminalLinkDetectorAdapter.ts

```diff
@@ -46,20 +46,16 @@ export class TerminalLinkDetectorAdapter extends Disposable implements ILinkProv
 	async provideLinks(bufferLineNumber: number, callback: (links: ILink[] | undefined) => void) {
 		let activeRequest = this._activeProvideLinkRequests.get(bufferLineNumber);
 		if (activeRequest) {
-			await activeRequest;
-			callback(this._activeLinks);
+			const links = await activeRequest;
```

Comment:
It's bit strange we used `this._activeLinks` instead of result of `await activeRequest` (which is `Promise<TerminalLinks[]` before. 

By using `callback(links)` we won't need `_activeLinks` anymore and can just rely on `activeLinksStore`

Reply:
Your proposal to use the return value does seem more correct, just make sure it's tested well